// lodash-clean/lib/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.buildCleaner = buildCleaner;
Object.defineProperty(exports, "fieldTypeCleaners", {
  enumerable: true,
  get: function get() {
    return _munge.fieldTypeCleaners;
  }
});
Object.defineProperty(exports, "buildGetValue", {
  enumerable: true,
  get: function get() {
    return _munge.buildGetValue;
  }
});
Object.defineProperty(exports, "getValue", {
  enumerable: true,
  get: function get() {
    return _munge.getValue;
  }
});
exports.default = exports.cleaner = void 0;

var _isFunction2 = _interopRequireDefault(require("lodash/fp/isFunction"));

var _munge = require("./munge");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function getMungeFunc(mungeOption) {
  if ((0, _isFunction2.default)(mungeOption)) return mungeOption(_munge.fieldTypeCleaners);
  return (0, _munge.buildGetValue)(mungeOption);
}

function buildCleaner(mungeOption) {
  var cleaner = getMungeFunc(mungeOption);
  return function (node) {
    return cleaner(node, cleaner);
  };
}

var cleaner = function cleaner(obj) {
  return (0, _munge.getValue)(obj, _munge.getValue);
};

exports.cleaner = cleaner;
var _default = cleaner;
exports.default = _default;
